lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -287,12 +287,18 @@ def add_files_to_build_phases(native_target, test_native_targets, app_native_tar
               {
                 true => file_accessor.arc_source_files,
                 false => file_accessor.non_arc_source_files,
-              }.each do |arc, files|
-                next if files.empty?
-                files = files - headers - other_source_files
-                flags = compiler_flags_for_consumer(consumer, arc)
-                regular_file_refs = project_file_references_array(files, 'source')
-                native_target.add_file_references(regular_file_refs, flags)
+              }.each do |arc, source_files|
+                next if source_files.empty?
+                source_files = source_files - headers - other_source_files
+                swift_source_files, non_swift_source_files = source_files.partition { |file| file.extname == '.swift' }
+
+                non_swift_compiler_flags = compiler_flags_for_consumer(consumer, arc, :objc)
```

Comment:
seems fine, would be nice to do this in a loop since we're duplicating this 3 lines of logic here

Reply:
done